### ISSUE TITLE
Assertion from CLI when querying unbound variable

### ISSUE THREAD

**Issue by thomcc** [mozilla/mentat#518]
Specifically [this assert](https://github.com/mozilla/mentat/blob/158910484191dee013b299d130f48078f376119b/query-projector/src/lib.rs#L211) is hit.

The following sequence triggers it in the CLI
```
.t [{:db/ident :test/name :db/valueType :db.type/string :db/cardinality :db.cardinality/one}]
.t [{:test/name "foo"}]
.q [:find ?eid ?v :where [65537 _ ?v]]
```

The log of the queries made, specific error message, and backtrace is available here: https://gist.github.com/thomcc/e19e15341b0aa4b461ceba1a972b5bd8

**Comment by rnewman** [comment]
This is because we're not verifying that the set of projected variables is a subset of the variables mentioned in the query body unioned with bound inputs. 

We can fix this either by extending `run_algebrized_query`, or by adjusting the projector's `candidate_column` function to be fallible.

**Comment by rnewman** [comment]
Closed by #590.
